tools/issue-labeler/src/IssueLabelerService/Answers/RagQna.cs

```diff
@@ -0,0 +1,146 @@
+using System;
+using System.Threading.Tasks;
+using IssueLabeler.Shared;
+using Newtonsoft.Json;
+using Microsoft.Extensions.Logging;
+using System.Linq;
+using System.Collections.Generic;
+
+namespace IssueLabelerService
+{
+    public class RagQna : IQnaService
+    {
+        private RepositoryConfiguration _config;
+        private TriageRag _ragService;
+        private ILogger<QnaFactory> _logger;
+        public RagQna(ILogger<QnaFactory> logger, RepositoryConfiguration config, TriageRag ragService)
+        {
+            _config = config;
+            _ragService = ragService;
+            _logger = logger;
+        }
+        public async Task<QnaOutput> AnswerQuery(IssuePayload issue)
+        {
+            // Configuration for Azure services
+            var modelName = _config.OpenAIModelName;
+            var issueIndexName = _config.IssueIndexName;
+            var documentIndexName = _config.DocumentIndexName;
+
+            // Issue specific configurations
+            var issueSemanticName = _config.IssueSemanticName;
+            string issueFieldName = _config.IssueIndexFieldName;
+
+            // Document specific configurations
+            var documentSemanticName = _config.DocumentSemanticName;
+            string documentFieldName = _config.DocumentIndexFieldName;
+
+            // Query + Filtering configurations
+            string query = $"{issue.Title} {issue.Body}";
+            int top = int.Parse(_config.SourceCount);
+            double scoreThreshold = double.Parse(_config.ScoreThreshold);
+            double solutionThreshold = double.Parse(_config.SolutionThreshold);
+
+            var relevantIssues = await _ragService.AzureSearchQueryAsync<Issue>(issueIndexName, issueSemanticName, issueFieldName, query, top);
+            var relevantDocuments = await _ragService.AzureSearchQueryAsync<Document>(documentIndexName, documentSemanticName, documentFieldName, query, top);
+
+            // Filter sources under threshold
+            var docs = relevantDocuments
+                .Where(r => r.Item2 >= scoreThreshold)
+                .Select(rd => new
+                {
+                    rd.Item1.chunk,
+                    rd.Item1.Url,
+                    Score = rd.Item2
+                })
+                .ToList();
+
+            var issues = relevantIssues
+                .Where(r => r.Item2 >= scoreThreshold)
+                .Select(rd => new
+                {
+                    rd.Item1.Title,
+                    rd.Item1.chunk,
+                    rd.Item1.Service,
+                    rd.Item1.Category,
+                    rd.Item1.Url,
+                    Score = rd.Item2
+                })
+                .ToList();
+
+            // Filtered out all sources for either one then not enough information to answer the issue. 
+            if (docs.Count() == 0 || issues.Count() == 0)
```

Comment:
```suggestion
            if (docs.Count == 0 || issues.Count == 0)
```

For a list, `Count` is a property that just reads a backing field.   `Count()` is a LINQ extension method that you should always expect to walk the entire enumerable.    If you don't have a known enumerable type, `Any()` would be better than `Count()` for asking "is there anything in there?"    In this case, we know it's a list, so we can go direct.